### ISSUE TITLE
install_deps(pkgdir, upgrade = "ask") doesn't respect locked versions in pkg DESCRIPTION

### ISSUE THREAD

**Issue by d-sci** [r-lib/remotes#443]
I have `install_deps(".")` set in my .Rprofile when working on/with my package so as to always have all/the latest dependencies. However, I have some of my package dependency versions locked down for safety. In my DESCRIPTION file:

```
Imports:
    ...,
    data.table (== 1.12.2),
    ...,
```

But then whenever I start up R, `install_deps` asks me:

```
These packages have more recent versions available.
Which would you like to update?

1: All                                 
2: CRAN packages only                  
3: None                                
4: data.table (1.12.2 -> 1.12.4) [CRAN]

Enter one or more numbers, or an empty line to skip updates:
```

Clearly I would NOT like data.table to upgrade, but I would like anything else. Can `install_deps` be smart enough to parse the version specs in my DESCRIPTION file? I want to be able to run this confidently in a non-interactive session (with upgrade = "always"), and I don't want to see this every time I load up R if no other package has updated.

**Comment by jimhester** [comment]
Having locked dependencies is basically unworkable with just R packages, mainly because only one package version can be loaded at the same time. You really need to use something that manages a private project library like [renv](https://github.com/rstudio/renv) if you want control over the exact versions used.

**Comment by d-sci** [comment]
Fair point. Thanks for the link to `renv` - I had given up on `packrat` but maybe this is worth another shot.

**Comment by rreitmann** [comment]
IMHO there is one use case in which `install_deps` should respect locked versions:
When running containerized CI builds (like on TravisCI) the build starts without any preinstalled R packages. In order to have reproducible builds I would like to use a fixed R version as well as fixed dependency versions.
